### DISCUSSION TITLE
Budget setting issue 500 Error

### DISCUSSION THREAD

**Discussion by mbobbar** [firefly-iii/firefly-iii#8942] [Firefly III]
Refering to issue; https://github.com/firefly-iii/firefly-iii/issues/8941#issuecomment-2143721224

I have checked my installation for the suggested commit changes, and believe its already in a 'fixed' state.
I understand i should be adding in '-->nullable()', is my installation correct as below?
![image](https://github.com/firefly-iii/firefly-iii/assets/142314522/ffe09acf-6b03-47b5-b999-8369c2824ca2)

**Comment by JC5** [comment]
Mmm, good addition! It should be correct, but if the error is still there, the column may not in fact be "nullable".

You're using Docker with sqlite, do you have the database stored somewhere accessible?

**Reply by mbobbar** [reply]
Yes, running as a home assistant OS add-on.
I actually didnt make any changes to the .php, hence the follow up to confirm why I'm still having issues!
Can confirm the database shows not null as below, should it be changed?

![Screenshot_20240602_190155_Chrome.jpg](https://github.com/firefly-iii/firefly-iii/assets/142314522/d7b290e8-5330-4e43-a44a-a885f7fdfb85)

Thanks for the help @JC5 👍

**Reply by JC5** [reply]
Yes, that field should be nullable. I am not sure though, how that works in sqlite. Tread with care!

**Reply by mbobbar** [reply]
Thanks! It restores functionality after deleting the database and restarting. Though I'm not sure how to proceed without data loss, I'm glad I have a fix.
Down the rabbit hole I go..

**Reply by mbobbar** [reply]
I've cleared the NOT NULL flag from the field by using a windows SQLite editor, it seems to have fixed the issue!
